src/rgw/rgw_rest_iam.cc

```diff
@@ -18,78 +18,48 @@
 
 using namespace std;
 
-void RGWHandler_REST_IAM::rgw_iam_parse_input()
+static const std::unordered_map<std::string, std::function<RGWOp*(const bufferlist&)>> op_generators = {
+  {"CreateRole", [](const bufferlist& bl_post_body) {return new RGWCreateRole(bl_post_body);}},
```

Comment:
i like this map-of-factories strategy 👍

all of the keys are string literals, so you can use std::string_view as the key type
lambdas with no capture can be converted to function pointers so std::function is unnecessary for the value type

Reply:
thanks!
i think i have to keep std::function. if I use a function pointer type, I get:
```
RGWOp*(const ceph::buffer::v15_2_0::list&)>::second’ invalidly declared function type
```
seems to be an issue, according to this: https://stackoverflow.com/questions/32332666/why-is-this-declaration-of-a-function-in-template-class-invalid